File: majority_element/majel.py
from typing import List

def majorityElement(nums: List[int]) -> int:
  half = len(nums) // 2
  numOccurrences = {}
  for num in nums:
    if (num in numOccurrences):
      numOccurrences[num] += 1
    else:
      numOccurrences[num] = 1
    if (numOccurrences[num] > half):
      return num;
  return nums[0] # for 2 size arrays

# nums = [3,2,3] # 3
# nums = [-1,-1,2147483647] # -1
# nums = [3,2] # 3
nums = [3] # 3

res = majorityElement(nums)

print(res)

File: majority_element/majel.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
  half = nums.size / 2
  num_occurrences = Hash.new(0)
  nums.each do |num|
    num_occurrences[num] += 1
    return num if (num_occurrences[num] > half)
  end
  nums[0] # for the 2 elements arrays
end

# nums = [2,2,1,1,1,2,2] # 2
# nums = [3,2,3] # 3
nums = [3,2] # 3
# nums = [2] # 2

res = majority_element(nums)
puts res.inspect
